tests/components/dlna_dms/test_config_flow.py

```diff
@@ -0,0 +1,346 @@
+from __future__ import annotations
+
+import dataclasses
+from typing import Final
+from unittest.mock import Mock
+
+from async_upnp_client import UpnpError
+import pytest
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components import ssdp
+from homeassistant.components.dlna_dms.const import DOMAIN
+from homeassistant.const import CONF_DEVICE_ID, CONF_HOST, CONF_URL
+from homeassistant.core import HomeAssistant
+
+from .conftest import (
+    MOCK_DEVICE_HOST,
+    MOCK_DEVICE_LOCATION,
+    MOCK_DEVICE_NAME,
+    MOCK_DEVICE_TYPE,
+    MOCK_DEVICE_UDN,
+    MOCK_DEVICE_USN,
+    NEW_DEVICE_LOCATION,
+)
+
+from tests.common import MockConfigEntry
+
+# Auto-use the domain_data_mock and dms_device_mock fixtures for every test in this module
+pytestmark = [
+    pytest.mark.usefixtures("domain_data_mock"),
+    pytest.mark.usefixtures("dms_device_mock"),
+]
+
+WRONG_DEVICE_TYPE: Final = "urn:schemas-upnp-org:device:InternetGatewayDevice:1"
+
+MOCK_ROOT_DEVICE_UDN: Final = "ROOT_DEVICE"
+
+MOCK_DISCOVERY: Final = ssdp.SsdpServiceInfo(
+    ssdp_usn=MOCK_DEVICE_USN,
+    ssdp_location=MOCK_DEVICE_LOCATION,
+    ssdp_udn=MOCK_DEVICE_UDN,
+    ssdp_st=MOCK_DEVICE_TYPE,
+    upnp={
+        ssdp.ATTR_UPNP_UDN: MOCK_ROOT_DEVICE_UDN,
+        ssdp.ATTR_UPNP_DEVICE_TYPE: MOCK_DEVICE_TYPE,
+        ssdp.ATTR_UPNP_FRIENDLY_NAME: MOCK_DEVICE_NAME,
+        ssdp.ATTR_UPNP_SERVICE_LIST: {
+            "service": [
+                {
+                    "SCPDURL": "/ContentDirectory/scpd.xml",
+                    "controlURL": "/ContentDirectory/control.xml",
+                    "eventSubURL": "/ContentDirectory/event.xml",
+                    "serviceId": "urn:upnp-org:serviceId:ContentDirectory",
+                    "serviceType": "urn:schemas-upnp-org:service:ContentDirectory:1",
+                },
+                {
+                    "SCPDURL": "/ConnectionManager/scpd.xml",
+                    "controlURL": "/ConnectionManager/control.xml",
+                    "eventSubURL": "/ConnectionManager/event.xml",
+                    "serviceId": "urn:upnp-org:serviceId:ConnectionManager",
+                    "serviceType": "urn:schemas-upnp-org:service:ConnectionManager:1",
+                },
+            ]
+        },
+    },
+    x_homeassistant_matching_domains={DOMAIN},
+)
+
+
+async def test_user_flow(hass: HomeAssistant, ssdp_scanner_mock: Mock) -> None:
+    """Test user-init'd flow, user selects discovered device."""
+    ssdp_scanner_mock.async_get_discovery_info_by_st.side_effect = [
+        [MOCK_DISCOVERY],
+        [],
+        [],
+        [],
+    ]
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] is None
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
We normally patch the integration setup function on all create entry results to avoid setting up the whole integration. See the scaffold script test template for example.